internal/providers/terraform/aws/autoscaling_group.go

```diff
@@ -35,6 +36,23 @@ func NewAutoscalingGroup(d *schema.ResourceData, u *schema.UsageData) *schema.Re
 
 	subResources := make([]*schema.Resource, 0)
 
+	// AWS Launch Template form terraform-aws-modules/eks/aws module
+	r := d.References("launch_template")
+	if r != nil {
+		onDemandCount, spotCount := desiredCapacity, decimal.Zero
+		if r[0].Get("instance_market_options.0.market_type").String() == "spot" {
+			onDemandCount = decimal.Zero
+			spotCount = desiredCapacity
+		}
+
+		lt := newLaunchTemplate(r[0].Address, r[0], u, region, onDemandCount, spotCount)
+		subResources = append(subResources, lt)
+		return &schema.Resource{
+			Name:         d.Address,
+			SubResources: subResources,
+		}
+	}
```

Comment:
I think we can't do it with current parser changes, because it doesn't create array of `launch_template` so `launch_template.0.*` doesn't work for this case

Reply:
I'm not sure I understand. When we call `.AddReference` it adds it as an array. I tried changing [this line](https://github.com/ekifel/infracost/blob/launch_template/internal/providers/terraform/parser.go#L601) to `Attribute:        "launch_template.0.id",` and it worked.

The other option would be to change L56-64 to something like:
```go
launchConfigurationRef := d.References("launch_configuration")

launchTemplateRef := d.References("launch_template")
if len(launchTemplateRef) == 0 {
	launchTemplateRef = d.References("launch_template.0.id")
}
if len(launchTemplateRef) == 0 {
	launchTemplateRef = d.References("launch_template.0.name")
}
```

Do you think this would work?

Reply:
@aliscott, yes, it works. Looks good, I added it